host_core/lib/host_core/actors/actor_supervisor.ex

```diff
@@ -203,49 +216,115 @@ defmodule HostCore.Actors.ActorSupervisor do
     end
   end
 
+  def start_actor_from_file(host_id, fileref, count \\ 1, annotations \\ %{}) do
+    {:ok, {pid, _prefix}} = VirtualHost.lookup(host_id)
+    config = VirtualHost.config(pid)
```

Comment:
I know this came in after you created this PR, but you can actually avoid the lookup and instead just get the config by `host_id` now to save a possibly blocking call